scripts/sorting.py

```diff
@@ -59,20 +61,37 @@ def _run_sorting_job(algorithm: str, recording_nwb_uri: str, sorting_params: dic
             'sorting_npz_uri': sorting_npz_uri
         }
 
+def _run_sorting_jobs_wrapper(job: Job, **kwargs):
+    print(f'Running: {job.label}')
+    output = _run_sorting_job(**job.kwargs, **kwargs)
+    print(f'OUTPUT of {job.label}:\n{output}')
+    kc.set(job.key(), output)
+
+
 @click.command()
 @click.argument('config_file')
 @click.argument('algorithm')
-@click.option('--force-run', is_flag=True, help="Force rerun")
+@click.option('--max-simultaneous-sorts', help="Maximum number of sorting jobs to run simultaneously")
+@click.option('--force-run', is_flag=True, help="Force rerurn")
 @click.option('--rerun-failing', is_flag=True, help="Rerun the failing jobs")
 @click.option('--docker', is_flag=True, help="Use docker image")
 @click.option('--singularity', is_flag=True, help="Use singularity image")
 @click.option('--image', default=None, help='Image for use in docker or singularity mode')
-def main(config_file: str, algorithm: str, force_run: bool, rerun_failing: bool, docker: bool, singularity: bool, image: Union[str, None]):
+def main(config_file: str, algorithm: str, max_simultaneous_sorts: Union[int, None], force_run: bool, rerun_failing: bool, docker: bool, singularity: bool, image: Union[str, None]):
+    if docker and singularity:
+        raise Exception('Both singularity and docker were requested, but no more than one can be used simultaneously.')
+    try:
+        max_simultaneous_sorts = int(max_simultaneous_sorts)
+        if max_simultaneous_sorts < 1:
+            max_simultaneous_sorts = 1
+    except:
+        max_simultaneous_sorts = 1
     with open(config_file, 'r') as f:
         config = yaml.safe_load(f)
     config_name = config['name']
     jobs0 = kc.get({'type': 'spikeforest-workflow-jobs', 'name': config_name})
     jobs: List[Job] = [Job.from_dict(job0) for job0 in jobs0]
+    #### TODO: Should this 'algorithm' actually be 'name'?
```

Comment:
No, algorithm is correct.

Reply:
This is somewhat confusing, because the same algorithm can appear many times in the `config.yaml` with different parameters, distinguished by the`name` key (as we said to Alessio and Sam). I do see in `workflow.py` that the `sorting_params` are copied into the Job dict, so they will be separate, but one could imagine a case where the user has requested 5x as many runs of (e.g.) MountainSort as any other algorithm, because they're testing different sets of sorting parameters. That would impact scheduling.

It also seems kinda weird that we go to some effort to provide granular control at the algorithmic level but require running all parameter sets for a given sorter.